### PR TITLE
DSND-2549: GET company officers incorrectly returns 404 when no officers exist for company

### PR DESCRIPTION
**Pull request by crowlandsCH**
* Call to metrics-api in CompanyAppointmentsService executed 
after call to delta_appointments collection.
* Implementation follows table on linked Jira ticket comments. 
* 503 error messages should no longer be spammed on frontend. 
* Frontend tested locally, shows up the same as the Perl.
* Additional integration test cases added for better coverage.

Locally tested and compared with the Perl, works as similarly as can get without changing the SDK handler. Please see tables on the Jira ticket. 

[DSND-2549](https://companieshouse.atlassian.net/browse/DSND-2549)

[DSND-2549]: https://companieshouse.atlassian.net/browse/DSND-2549?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/it/java/uk/gov/companieshouse/company_appointments/tests/CompanyAppointmentControllerITest.java [hunk 1]
```diff
@@ -187,7 +189,17 @@ void testReturn404IfOfficersForCompanyIsNotFound() throws Exception {
...
+        result.andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.total_results", is(0)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.items", is(new ArrayList<>())))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.active_count", is(0)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.inactive_count", is(0)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.resigned_count", is(0)))
+                .andExpect((MockMvcResultMatchers.jsonPath("$.kind", is("officer-list"))))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.start_index", is(0)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.items_per_page", is(35)))
+//                .andExpect(MockMvcResultMatchers.jsonPath("links", is)
```
Comment (johncook-ch) [inline src/it/java/uk/gov/companieshouse/company_appointments/tests/CompanyAppointmentControllerITest.java:201]: Remove comment out line

File: src/it/java/uk/gov/companieshouse/company_appointments/tests/CompanyAppointmentControllerITest.java [hunk 2]
```diff
@@ -175,7 +177,7 @@ void testReturn200OKIfAllOfficersAreFound() throws Exception {
                 .andExpect(MockMvcResultMatchers.jsonPath("$.total_results", is(3)));
     }
     @Test
-    void testReturn404IfOfficersForCompanyIsNotFound() throws Exception {
+    void testReturnBaseResponseIfOfficersForCompanyIsNotFound() throws Exception {
```
Comment (johncook-ch) [inline src/it/java/uk/gov/companieshouse/company_appointments/tests/CompanyAppointmentControllerITest.java:180]: The name doesn't tell much. How about `shouldReturnZeroCountsWhenOfficerForCompanyIsNotFound()`

File: src/test/java/uk/gov/companieshouse/company_appointments/service/CompanyAppointmentServiceTest.java [hunk 3]
```diff
@@ -435,7 +437,7 @@ void testFetchAppointmentsForCompanyThrowsBadRequestExceptionIfRegisterViewAndIn
     }
 
     @Test
-    void testFetchAppointmentsForCompanyThrowsNotFoundExceptionIfNoCountsInMetrics()
+    void testFetchAppointmentsForCompanyReturnsBaseResponseIfNoCountsOrNullCountsInMetrics()
```
Comment (johncook-ch) [inline src/test/java/uk/gov/companieshouse/company_appointments/service/CompanyAppointmentServiceTest.java:440]: Ass above, 'BaseDocument' won't mean to much to a furure developer, so the name should be clearer

Comment (crowlandsCH) [reply src/test/java/uk/gov/companieshouse/company_appointments/service/CompanyAppointmentServiceTest.java]: addressed in most new commit

File: src/it/java/uk/gov/companieshouse/company_appointments/tests/CompanyAppointmentControllerITest.java (hunk 2 above)
Comment (crowlandsCH) [reply src/it/java/uk/gov/companieshouse/company_appointments/tests/CompanyAppointmentControllerITest.java]: addressed in most new commit

File: src/it/java/uk/gov/companieshouse/company_appointments/tests/CompanyAppointmentControllerITest.java (hunk 1 above)
Comment (crowlandsCH) [reply src/it/java/uk/gov/companieshouse/company_appointments/tests/CompanyAppointmentControllerITest.java]: addressed in most new commit

File: src/test/java/uk/gov/companieshouse/company_appointments/api/CompanyMetricsApiServiceTest.java [hunk 4]
```diff
@@ -97,4 +98,17 @@ void whenApiThrowsExceptionThenThrowServiceUnavailableException() throws Excepti
         assertThrows(ServiceUnavailableException.class,
                 () -> service.invokeGetMetricsApi(COMPANY_NUMBER));
     }
+
+    @Test
+    void whenApiReturns404StatusThenReturn200OK() throws ApiErrorResponseException, URIValidationException {
```
[the remark this answers is not in the document]
Comment (crowlandsCH) [reply src/test/java/uk/gov/companieshouse/company_appointments/api/CompanyMetricsApiServiceTest.java]: done in most recent commit

**Review by johncook-ch — APPROVED** [review]

**Pull request merged**